mitodl/mitxonline
# Fixes some issues with validation for new profile fields; adds extended profile fields

#### Pre-Flight checklist


#### What are the relevant tickets?

Related to #1399 and #1425 and #1424

#### What's this PR do?

There were some bugs in validation and saving for users that had existing accounts:
- In Edit Profile, the Gender and Year of Birth fields weren't validating at all
- Saving with these extra fields caused an error if the user didn't already have a UserProfile record (which they won't if they're old)

In addition, this adds the extra fields and updates the tests and serializers for #1424. All the fields are nullable and I've cleaned up the UserProfileSerializer a bit too. (I wanted to get these fixes out before being fully done with the UI stuff since it's blocking the release of the minimal fields from #1399.)

#### How should this be manually tested?

New user registration should work as it has.

As an existing user that _does not_ have an existing UserProfile record, edit and save your profile. You should get a validation error if you leave the Year of Birth field blank, and the data should save successfully. 

As an existing user that _did have_ an existing UserProfile record with a blank Year of Birth field, test editing the profile. Validation should work as expected. Your UserProfile record should have the Year of Birth field set to `3000` (which should ensure it fails validation).
